app/renderer/js/pages/preference/general-section.ts

```diff
@@ -493,4 +499,59 @@ export default class GeneralSection extends BaseSection {
 			await this.factoryResetSettings();
 		});
 	}
+
+	initSpellChecker(): void {
+		// The elctron API is a no-op on macOS and macOS default spellchecker is used.
+		if (process.platform === 'darwin') {
+			const note = document.querySelector('#note');
+			note.innerHTML += 'On macOS, the OS spellchecker is used.<br>Change the language from System Preferences -> Keyboard -> Text -> Spelling.';
+		} else {
+			const note = document.querySelector('#note');
+			note.innerHTML += 'You can select a maximum of 3 languages for spellchecking.';
+			const spellDiv = document.querySelector('#spellcheck-langs');
+			spellDiv.innerHTML += `
+				<div class="setting-description">${t.__('SpellChecker Languages')}</div>
+				<input name='spellcheck' placeholder='Enter Languages'>`;
+
+			const availableLanguages = session.fromPartition('persist:webviewsession').availableSpellCheckerLanguages;
+			const languagePairs: Map<string, string> = new Map();
+			availableLanguages.forEach((l: string) => {
+				if (ISO6391.getName(l) !== '') {
+					languagePairs.set(ISO6391.getName(l), l);
+				}
+			});
+
+			// Manually set names for languages not available in ISO6391
+			languagePairs.set('English (AU)', 'en-AU'); languagePairs.set('English (CA)', 'en-CA'); languagePairs.set('English (GB)', 'en-GB'); languagePairs.set('English (US)', 'en-US'); languagePairs.set('Spanish (Latin America)', 'es-419'); languagePairs.set('Spanish (Argentina)', 'es-AR'); languagePairs.set('Spanish (Mexico)', 'es-MX'); languagePairs.set('Spanish (US)', 'es-US'); languagePairs.set('Portuguese (Brazil)', 'pt-BR'); languagePairs.set('Portuguese (Portugal)', 'pt-PT'); languagePairs.set('Serbo-Croatian', 'sh'); // eslint-disable-line max-statements-per-line
+			const tagField = document.querySelector('input[name=spellcheck]');
+			const tagify = new Tagify(tagField, {
+				whitelist: [...languagePairs.keys()],
+				enforceWhitelist: true,
+				maxTags: 3,
+				dropdown: {
+					enabled: 0,
+					maxItems: 100, // If chromium supports more and more languages in future
+					closeOnSelect: false,
+					highlightFirst: true
+				}
+			});
+
+			const configuredLanguages: string[] = [];
+			ConfigUtil.getConfigItem('spellcheck-languages').forEach((code: any) => {
+				[...languagePairs].forEach((pair): string | void => {
+					if (pair[1] === code) {
+						configuredLanguages.push(pair[0]);
+					}
+				});
+			});
```

Comment:
This is a quadratic loop that could be avoided with better data structures.

`string | void` doesn’t make sense.

Generally loops around `push` should be avoided in favor of `map`.